src/main/java/com/rybka/service/ExchangeService.java

```diff
@@ -1,100 +1,61 @@
-            System.exit(3);
+    public ExchangeRate loadCurrencyOf(String userBaseCurrency, String userTargetCurrency) throws NullResponseException, ZeroCurrencyRateException {
+        var response = new ApiConnectorService().retrieveRates(userBaseCurrency);
+        var currencyRate = retrieveJsonData(response, userTargetCurrency);
+        if (currencyRate == 0.0) {
+            throw new ZeroCurrencyRateException("Invalid currency");
         }
+        return new ExchangeRate(userBaseCurrency, userTargetCurrency, currencyRate);
     }
 
-    public void exchange(Double amount) {
-
-        this.amount = amount;
-        this.total = amount * currencyValue;
-    }
-
-    public void getCurrencyObject() {
-        Currency currency = new Currency(baseCurrencyAbbreviation, targetCurrencyAbbreviation, currencyValue, amount, total);
-        HibernateDAO hibernateDAO = new HibernateDAO();
-        hibernateDAO.save(currency);
-        hibernateDAO.showTableRow();
-    }
-
-    public Double getCurrencyValue() {
-        return currencyValue;
-    }
-
-    public void setCurrencyValue(Double currencyValue) {
-        this.currencyValue = currencyValue;
+    public ConvertedCurrency calculateTotal(ExchangeRate exchangeRate, Double amount) {
+        ConvertedCurrency currency = new ConvertedCurrency();
+        currency.setBase(exchangeRate.getBase());
+        currency.setTarget(exchangeRate.getTarget());
+        currency.setRate(exchangeRate.getRate());
+        currency.setAmount(amount);
+        currency.setTotal(currency.getAmount() * currency.getRate());
+        return currency;
     }
 
-    public String getBaseCurrencyAbbreviation() {
-        return baseCurrencyAbbreviation;
-    }
-
-    public void setBaseCurrencyAbbreviation(String baseCurrencyAbbreviation) {
-        this.baseCurrencyAbbreviation = baseCurrencyAbbreviation;
-    }
-
-    public String getTargetCurrencyAbbreviation() {
-        return targetCurrencyAbbreviation;
-    }
-
-    public void setTargetCurrencyAbbreviation(String targetCurrencyAbbreviation) {
-        this.targetCurrencyAbbreviation = targetCurrencyAbbreviation;
-    }
-
-    public Double getAmount() {
-        return amount;
-    }
-
-    public void setAmount(Double amount) {
-        this.amount = amount;
-    }
-
-    public Double getTotal() {
-        return total;
-    }
-
-    public void setTotal(Double total) {
-        this.total = total;
+    private Double retrieveJsonData(HttpResponse<String> response, String userTargetCurrency) {
+        ObjectMapper mapper = new ObjectMapper();
+        var currencyRate = 0.0;
+        try {
+            JsonNode rootNode = mapper.readTree(response.body());
+            var nameNode = rootNode.path("conversion_rates").toString();
+            TypeReference<HashMap<String, Double>> typeRef
+                    = new TypeReference<>() {
+            };
+            Map<String, Double> map = mapper.readValue(nameNode, typeRef);
```

Comment:
I'm not sure we to do this filtering here. We can do this when retrieve all rates for the base currency.

Reply:
fixed